esp/esdllib/esdl_script.cpp

```diff
@@ -234,6 +239,416 @@ interface IEsdlTransformOperationHttpHeader : public IInterface
+}
+
+static inline void addExceptionsToXpathContext(IXpathContext *targetContext, IMultiException *me)
+{
+    if (!targetContext || !me)
+        return;
+    StringBuffer xml;
+    me->serialize(xml);
+    CXpathContextLocation content_location(targetContext);
+    targetContext->ensureSetValue("@status", "error", true);
+    targetContext->addXmlContent(xml.str());
+}
+
+static inline void addExceptionsToXpathContext(IXpathContext *targetContext, IException *E)
+{
+    if (!targetContext || !E)
+        return;
+    Owned<IMultiException> me = makeMultiException("ESDLScript");
+    me->append(*LINK(E));
+    addExceptionsToXpathContext(targetContext, me);
+}
+
+class CEsdlTransformOperationMySqlCall : public CEsdlTransformOperationBase
+{
+protected:
+    StringAttr m_name;
+
+    Owned<ICompiledXpath> m_vaultName;
+    Owned<ICompiledXpath> m_secretName;
+    Owned<ICompiledXpath> m_section;
+    Owned<ICompiledXpath> m_resultDelimiter;
+    Owned<ICompiledXpath> m_server;
+    Owned<ICompiledXpath> m_user;
+    Owned<ICompiledXpath> m_password;
+    Owned<ICompiledXpath> m_database;
+
+    StringArray m_mysqlOptionNames;
+    IArrayOf<ICompiledXpath> m_mysqlOptionXpaths;
+
+    StringBuffer m_sql;
+
+    Owned<ICompiledXpath> m_select;
+
+    IArrayOf<CEsdlTransformOperationMySqlBindParmeter> m_parameters;
+
+public:
+    CEsdlTransformOperationMySqlCall(XmlPullParser &xpp, StartTag &stag, const StringBuffer &prefix) : CEsdlTransformOperationBase(xpp, stag, prefix)
+    {
+        ensureMysqlEmbeded();
+
+        m_name.set(stag.getValue("name"));
+        if (m_traceName.isEmpty())
+            m_traceName.set(m_name.str());
+
+        //select is optional, with select, a mysql call behaves like a for-each, binding and executing each iteration of the selected content
+        //without select, it executes once in the current context
+        m_select.setown(compileOptionalXpath(stag.getValue("select")));
+
+        m_vaultName.setown(compileOptionalXpath(stag.getValue("vault")));
+        m_secretName.setown(compileOptionalXpath(stag.getValue("secret")));
+        m_section.setown(compileOptionalXpath(stag.getValue("section")));
+        m_resultDelimiter.setown(compileOptionalXpath(stag.getValue("result-delimiter")));
+
+        m_server.setown(compileOptionalXpath(stag.getValue("server")));
+        m_user.setown(compileOptionalXpath(stag.getValue("user")));
+        m_password.setown(compileOptionalXpath(stag.getValue("password")));
+        m_database.setown(compileOptionalXpath(stag.getValue("database")));
+
+        //script can set any MYSQL options using an attribute with the same name as the option enum, for example
+        //    MYSQL_SET_CHARSET_NAME="'latin1'" or MYSQL_SET_CHARSET_NAME="$charset"
+        //
+        int attCount = stag.getLength();
+        for (int i=0; i<attCount; i++)
+        {
+            const char *attName = stag.getLocalName(i);
+            if (attName && strncmp("MYSQL_", attName, 6)==0)
```

Comment:
Is there a preference here to use the hasPrefix() function?

Reply:
Good call.  Not sure if I knew that function existed or not.  I'll use it.

Reply:
Now using hasPrefix().